pkg/providers/launchtemplate/suite_test.go

```diff
@@ -1696,6 +1696,63 @@ var _ = Describe("LaunchTemplates", func() {
 				})
 			})
 		})
+		Context("Windows Custom UserData", func() {
+			BeforeEach(func() {
+				ctx = settings.ToContext(ctx, test.Settings(test.SettingOptions{
+					EnableENILimitedPodDensity: lo.ToPtr(false),
+				}))
+				provisioner.Spec.Requirements = []v1.NodeSelectorRequirement{{Key: v1.LabelOSStable, Operator: v1.NodeSelectorOpIn, Values: []string{"windows"}}}
+				nodeTemplate.Spec.AMIFamily = &v1alpha1.AMIFamilyWindows2022
+			})
+			It("should merge and bootstrap with custom user data if valid XML user data", func() {
```

Comment:
Can we test all the XML parameters that are available with the merging of the userData?

Reply:
From what I can tell we can be testing the `Script`, `Detach`, and `Persist` tags

Reply:
Removed the flexibility to specify EC2Launch XML elements. In the future if the need arises we can implement further changes to support this.